strategies/aartur/straregy.py

```diff
@@ -0,0 +1,50 @@
+import attr
+
+from strategy import BaseMove, BaseStrategy
+from rules import State
+
+from .preparation import AdvancedState, Map
+from .solver import BaseSolver, CollectHealBonusSolver, CollectScoreBonusSolver, ShootSolver, HideSolver, CenterSolver
+
+
+@attr.s(slots=True, kw_only=True)
+class AArturBaseStrategy(BaseStrategy):
+    """My base strategy! Chicken!
+
+    Stores some Solver objects and priority coefficients associated to them.
+
+    Calculating the resulting move is quite simple:
+    1. Call all our solvers -> moves and confidences
+    2. Multiply our priority to the given confidence
+    3. Pick a move with maximum value
+    """
+
+    solvers: dict[BaseSolver, float] = attr.ib(factory=dict, init=False)
+
+    def get_next_move(self, state: State) -> BaseMove:
+        moves: list[tuple[BaseMove, float]] = []  # if only the DirectMove and Shoot were hashable...
```

Comment:
for what need hash of DirectMove and Shoot?

Reply:
For more convinient usage such as using in `dict[BaseMove, something]` and checking the equality of moves.
Removed this comment